src/Microsoft.DotNet.Interactive.HttpRequestParser/Parser/HttpRequestNode.cs

```diff
@@ -81,6 +81,11 @@ public void Add(HttpBodyNode node)
         AddInternal(node);
     }
 
+    public void Add(HttpCommentNode node)
```

Comment:
Is this needed? So far, the explicit Add methods were only required for nodes that appear as properties...

Reply:
They're needed for all node types that can be added.

Reply:
We had discussed this and decided to include the Add method for comments on the base `HttpSyntaxNode` class (because otherwise implementing `ParseLeadingCommentsAndWhitespace` becomes cumbersome),

`HttpSyntaxNode` already has an `Add` method that handles comments. Looks like it has an additional `addBefore` parameter - perhaps we can make that parameter optional with a default value and remove the explicit one above?

Reply:
> We had discussed this and decided to include the Add method for comments on the base HttpSyntaxNode class (because otherwise implementing ParseLeadingCommentsAndWhitespace becomes cumbersome),

Since comments aren't valid in all syntax comments, adding this `Add` method on `HttpSyntaxNode` would be misleading.